### ISSUE TITLE
Support selection of capm3 and ipam pods for K8s service

### ISSUE THREAD

**Issue by prabhu43** [metal3-io/cluster-api-provider-metal3#2048]
**User Story**

As a developer I would like to add K8s Service for both capm3-controller-manager and ipam-controller-manager for deploying Prometheus ServiceMonitor to enable metrics collection

**Detailed Description**

As capm3-controller-manager and ipam-controller-manager pods have same labels, We could not select the pods with unique labels in Service.
```
      cluster.x-k8s.io/provider: infrastructure-metal3
      control-plane: controller-manager
      controller-tools.k8s.io: "1.0"
```

**Solution:**
- Add a new label "app: capm3" to capm3 resources
- Add a new label "app: ipam" to ipam resources

/kind feature

**Comment by Rozzii** [comment]
Hmm this is a potential bug, we have to discuss this on community meeting. I am fine with accepting this.

PTAL: @lentzi90 @adilGhaffarDev

**Comment by lentzi90** [comment]
Yes this sounds like a bug. IPAM should not have the `cluster.x-k8s.io/provider: infrastructure-metal3` label. It is not an infrastructure provider.

I was curious how the Deployments are able to function since they also use label selectors. It turns out that the only reason it is currently working is because the pod template hash is added to the selector of the ReplicaSet. :see_no_evil:

**Comment by prabhu43** [comment]
@Rozzii @lentzi90 removing the existing labels in selectors will fail the helm upgrade. so shall we add the new label "app" to differentiate. let me know if this solution works. happy to contribute to this

**Comment by Rozzii** [comment]
I am not against having labels for the controllers just to identify the a specific type of controller instance. But I am also not up to date on what are the best practices. I think we have to plan this deprecation as you have mentioned @prabhu43 other communities OSS as well as downstream might rely on the current labels.

I would propose adding the new app labels or something like that then putting a deprecation notice on the incorrect IPAM label.
Would this be ok in your opinion @lentzi90 ?

**Comment by lentzi90** [comment]
Let's add a new label `app.kubernetes.io/name` from the [recommended labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/). It could have the value `cluster-api-provider-metal3` for CAPM3 and `ip-address-manager` for IPAM I think.

Removing the label is tricky indeed. We would need to tell users to delete the old deployment and create a new on upgrade. I would say that we can let it be there for now, at least as long as we are bundling IPAM with CAPM3. When/if we make IPAM installable separately we can think about removing the label from there.

**Comment by tuminoid** [comment]
I agree with Lennart's approach, although, `app.kubernetes.io/name` is kind of global label name, so we might actually conflict with some other `ip-address-manager` as that is quite generic term. Since its only a label, maybe should use something that identifies it as Metal3 IPAM, `ip-address-manager-metal3` or such?

**Comment by lentzi90** [comment]
That is a good point. I feel that we are quite well protected since this is all inside the capm3-system namespace.
Did we have some name already for what to call IPAM when used as a CAPI ip-address-manager? We could use that already. There is of course the drawback that it is then not the same as the name of the repo, but oh well :shrug:

**Comment by adilGhaffarDev** [comment]
We cannot remove `cluster.x-k8s.io/provider:` we have to change its value. For metal3-IPAM label should be like this: `cluster.x-k8s.io/provider: metal3-ipam` , because we are working towards making [Metal3 IPAM](https://github.com/metal3-io/ip-address-manager) a IPAM provider for CAPI. For now, as @lentzi90 suggested let's add a new label.  @peppi-lotta can add more context. 
There is a open PR in IPAM that is already fixing this label: https://github.com/metal3-io/ip-address-manager/pull/692/files

**Comment by Rozzii** [comment]
/remove-lifecycle stale
I think this still makes sense
/lifecycle frozen
FYI @peppi-lotta
